### PR TITLE
fix: create profiler specific unsupported regions

### PR DESCRIPTION
**Pull request by jswudi**
*Issue #, if available:* We should not automatically enable profiler feature in unsupported regions.

*Description of changes:* Add an unsupported regions list for profiler feature.

*Testing done:* Add unit test

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/test_estimator.py [hunk 1]
```diff
@@ -632,6 +632,31 @@ def test_framework_with_profiler_config_without_s3_output_path(time, sagemaker_s
     ]
 
 
+def test_framework_with_no_default_profiler_in_unsupported_region():
+    boto_mock = Mock(name="boto_session", region_name="cn-north-1")
```
Comment (ndodda-amazon) [inline tests/unit/test_estimator.py:636]: I don't think we should hardcode the region in the test. Let's just test that every region in the unsupported list passes this test.

```
from sagemaker.fwutils import PROFILER_UNSUPPORTED_REGIONS
...
@pytest.mark.parametrize("region", PROFILER_UNSUPPORTED_REGIONS)
def test_framework_with_no_default_profiler_in_unsupported_region(region):
    boto_mock = Mock(name="boto_session", region_name=region)
...
```

Comment (ndodda-amazon) [reply tests/unit/test_estimator.py:636]: That way this test isn't dependent on what exact regions aren't supported.

Comment (Edge1211) [reply tests/unit/test_estimator.py:636]: +1 and also when we remove `cn-north-1` from the list, we won't have to update the test this way

Comment (jswudi) [reply tests/unit/test_estimator.py]: Done.

**Review by ndodda-amazon — APPROVED** [review]
LGTM, thanks for fixing!

**Review by Edge1211 — APPROVED** [review]

**Review by icywang86rui — APPROVED** [review]

**Pull request merged**